lib/features/children/generosity_challenge/cubit/generosity_challenge_vpc_setup_cubit.dart

```diff
@@ -67,31 +66,18 @@ class GenerosityChallengeVpcSetupCubit
     try {
       final userData = _generosityChallengeRepository.loadUserData();
       final children = _retrieveChildrenData(userData);
-      var email = '';
-      try {
-        email = getIt<SharedPreferences>()
-                .getString(ChatScriptSaveKey.email.value) ??
-            '';
-      } catch (e) {
-        email = userData[ChatScriptSaveKey.email.value] as String;
-        if (email.isEmpty) {
-          LoggingInfo.instance.error(
-            'Failed to get sign up email in generosity challenge VPC check: $e',
-            methodName: '_handleVPC',
-          );
-        }
-      }
+      final email = GenerosityChallengeHelper.getChallengeEmail(userData);
```

Comment:
I'm still a bit confused on the logic. We're not asking for the email in the chat anymore right? So I think we should retrieve the email from the temp account first and only if that fails look at SharedPreferences (for users that started the challenge before the phone vs email changes).